Sources/Kitura/CodableRouter.swift

```diff
@@ -604,14 +607,33 @@ extension Router {
         }
     }
 
-    internal func parameterIsPresent(in route: String) -> Bool {
-        if route.contains(":") {
-            let paramaterString = route.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
-            let parameter = paramaterString.count > 0 ? paramaterString[1] : ""
-            Log.error("Erroneous path '\(route)', parameter ':\(parameter)' is not allowed. Codable routes do not allow parameters.")
+    func pathSyntaxIsValid(_ path: String, identifierExpected: Bool) -> Bool {
+        let identifierSupplied = path.contains(":")
+        switch (identifierExpected, identifierSupplied) {
+        case (false, true):
+            Log.error("Path '\(path)' is not allowed: Codable routes do not allow path parameters.")
+            return false
+        case (true, false):
+            Log.warning("Identifier expected for '\(path)'. :id will be automatically appended to the path.")
+            return true
+        case (true, true):
+            /// Precondition: Path is known to contain a : character
+            func pathHasSingleParamIdAsSuffix(_ path: String) -> Bool {
```

Comment:
It seems _weird_ to be declaring a function within a case statement... if this function has no value outside the block, couldn't the logic just be inlined?